res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,736 @@
+ * 2016-08-13 - v1.2 - Improved jog leds
+ ***********************************************************************
+ *                           GPL v2 licence
+ *                           --------------
+ * Reloop Beatmix controller script script 1.2 for Mixxx 2.1.0
+ * Copyright (C) 2016 Sébastien Blaisot
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ ***********************************************************************/
+////////////////////////////////////////////////////////////////////////
+// JSHint configuration                                               //
+////////////////////////////////////////////////////////////////////////
+/* global engine                                                      */
+/* global script                                                      */
+/* global print                                                       */
+/* global midi                                                        */
+////////////////////////////////////////////////////////////////////////
+
+// Global variables and declarations.
+// ========================================================
+var ReloopBeatmix24 = {};
+
+var RateRangeArray = [0.08, 0.10, 0.12, 0.16];
+
+// Timers & long press state
+var jogWheelTimers = [];
+var loadButtonTimers = [];
+var loadButtonLongPressed = [];
+var FxModeTimers = [];
+var FxModeLongPressed = [];
+
+// A variable to store previous value of some connectControls
+var previousValue = [];
+
+// Trax mode
+// 1 for playlist mode
+// 2 for track mode
+// 3 for preview mode
+var traxMode = 2;
+
+// Effects mode
+// 1 for single effect mode (1 effect controlled in each EffectUnit)
+// 2 for multi-effect mode (3 effects controlled in each EffectUnit)
+// SHIFT + long press on pitchbend +/- to change mode
+var FxMode = 1; // Single effect mode by default
+
+// Jog Led variables
+var JogRPM = 33.0 + 1/3, // Jog Wheel simulate a 33.3RPM turntable
+    RoundTripTime = 60.0 / JogRPM, // Time in seconds for a complete turn
+    JogLedNumber = 16, // number of leds (sections) on the jog wheel
+    JogBaseLed = 0x3f, // Midino of last led (we count backward to turn in the right side)
+    JogFlashWarningTime = 30, // number of seconds to slowly blink at the end of track
```

Comment:
probably -1 is a better way to disable it because of the <= comparator, 0 could be reach, but we will never have -1s left and timeleft will never be <= -1
I will test this and add a comment about that